syntax = "proto3";

package sigmadsp.backend_service;

message ChangeVolume {
  string cell_name = 1;
  double value = 2;
  bool relative = 3;
}

message LoadParameters {
  repeated string content = 1;
}

message ControlRequest {
  oneof command {
    ChangeVolume change_volume = 1;
    bool reset_dsp = 2;
    LoadParameters load_parameters = 3;
  }
}

message ControlResponse {
  bool success = 1;
  string message = 2;
}

service Backend {
  rpc control (ControlRequest) returns (ControlResponse);
}
